### PR TITLE
Small fixes for mac arm compilation and fix some tests

### PR DESCRIPTION
**Pull request by maxim-morozov**
#### What does this implement or fix?
The fix for the compilation for MAC arm architecture that doesn't support `__yield` instruction. The fix also covers some unit test failures.

#### Any other comments?
Comments inline of the PR with explanations

#### Checklist

<details>
  <summary>
   Checklist for code changes...
  </summary>
 
 - [ ] Have you updated the relevant docstrings, documentation and copyright notice?
 - [x] Is this contribution tested against [all ArcticDB's features](../docs/mkdocs/docs/technical/contributing.md)?
 - [ ] Do all exceptions introduced raise appropriate [error messages](https://docs.arcticdb.io/error_messages/)?
 - [ ] Are API changes highlighted in the PR description?
 - [ ] Is the PR labelled as enhancement or bug so it appears in autogenerated release notes?
</details>

Signed-Off By Maxim Morozov [maxim.morozov@gmail.com](mailto:maxim.morozov@gmail.com). By including this sign-off line I agree to the terms of the Contributor License Agreement

### CODE REVIEW

File: cpp/arcticdb/entity/key.hpp [hunk 1]
```diff
@@ -256,12 +256,6 @@ inline KeyType get_key_type_for_index_stream(const StreamId &) {
     return KeyType::TABLE_INDEX;
 }
 
-template <typename Function>
-auto foreach_key_type(Function&& func) {
-    for(int k = 0; k < int(KeyType::UNDEFINED); ++k) {
```
Comment (maxim-morozov) [inline cpp/arcticdb/entity/key.hpp:261]: It looks like the keytype = 0 is not used and also there is already a function to convert int to KeyType

File: cpp/arcticdb/entity/metrics.hpp [hunk 2]
```diff
@@ -133,9 +133,8 @@ class PrometheusInstance {
             return;
         }
 
-        metrics.erase({name, labels});
-        metrics_family.erase(it);
```
Comment (maxim-morozov) [inline cpp/arcticdb/entity/metrics.hpp:137]: It looks like the remove operation also removes metrics-family though metrics-family might have additional metrics

File: cpp/arcticdb/storage/test/test_storage_exceptions.cpp [hunk 3]
```diff
@@ -202,8 +202,8 @@ class LMDBStorageTestBase : public ::testing::Test {
 };
 
 TEST_F(LMDBStorageTestBase, WriteMapFullError) {
-    // Create a Storage with 32KB map size
-    LMDBStorageFactory factory(32ULL * (1ULL << 10), false);
+    // Create a Storage with 64KB map size
+    LMDBStorageFactory factory(64ULL * (1ULL << 10), false);
```
Comment (maxim-morozov) [inline cpp/arcticdb/storage/test/test_storage_exceptions.cpp:206]: In the M1 arm environment, the 32KB map throws a map-full exception on allocation. Looks like the minimum is 64KB.

File: cpp/arcticdb/util/spinlock.hpp [hunk 4]
```diff
@@ -14,7 +14,7 @@ namespace arcticdb {
 #ifdef WIN32
 #define PAUSE _mm_pause()
 #elif defined(__arm__) || defined(__ARM_ARCH)
-#define PAUSE __yield()
+#define PAUSE asm volatile("yield")
```
Comment (maxim-morozov) [inline cpp/arcticdb/util/spinlock.hpp:17]: __yield is not supported on Mac arm architecture.

File: cpp/arcticdb/entity/key.hpp (hunk 1 above)
Comment (willdealtry) [reply cpp/arcticdb/entity/key.hpp:261]: Yes, KeyTypes 1 and 2 were part of the original and now entirely superseded purpose or ArcticDB (as node storage for a specific graph strategy) and were never written anywhere.

**Review by willdealtry — APPROVED** [review]

**Comment by maxim-morozov** [comment]
> Hi @maxim-morozov . Please include the contributor sign-off in the commit message.

Thanks @jamesmunro. I've included the contributor sign-off in the commit message as well.

**Review by poodlewars — APPROVED** [review]

**Pull request merged**